src/lib/duration/valid.js

```diff
@@ -0,0 +1,24 @@
+var ordering = ['year', 'quarter', 'month', 'week', 'day', 'hour', 'minute', 'second', 'millisecond'];
+
+export default function isDurationValid(m) {
+    for (var key in m) {
+        if (ordering.indexOf(key) === -1 ||
+        m[key] !== undefined && isNaN(parseInt(m[key]))) {
```

Comment:
Rewrite to:
`if (!(ordering.indexOf(key) !== -1 && (m[key] == null || !isNaN(m[key]))))`
The tricky part is -- I want to explicitly allow both null and undefined (they'll be treated as `0` because of the `||  0` code later). Also you don't need parseInt when doing isNaN (its built in).

I hope we can have some stricter rules in a future major release, but right now even this is getting dangerous (as it will break someone's code).